lib/tests/streamlit/elements/doc_string_test.py

```diff
@@ -33,6 +33,20 @@ def patch_varname_getter():
     )
 
 
+class ConditionalHello:
+    def __init__(self, available):
+        self.available = available
+
+    def __getattribute__(self, name):
+        if name == "say_hello" and not self.available:
+            raise AttributeError(f"{name} is not accessible when x is even")
```

Comment:
nit: the error message here seems unrelated to how we actually decide whether to throw an error in this method